From e5afa4352f69e66d9d27fd47ca6a8f759fa909ca Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@st.com>
Date: Mon, 10 Dec 2018 17:36:27 +0100
Subject: [PATCH 8/9] waylandsink: do not destroy pool twice

Fix segfault in wayland client due to pool
destroy being called with null pool...
This problem was encountered with GStreamer-1.14.

Signed-off-by: Hugues Fruchet <hugues.fruchet@st.com>
---
 ext/wayland/waylandpool.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ext/wayland/waylandpool.c b/ext/wayland/waylandpool.c
index 1ba1f85..17a94f1 100644
--- a/ext/wayland/waylandpool.c
+++ b/ext/wayland/waylandpool.c
@@ -417,7 +417,12 @@ gst_wayland_buffer_pool_stop (GstBufferPool * pool)
 
   GST_DEBUG_OBJECT (self, "Stopping wayland buffer pool");
 
+  /* already stopped... */
+  if (!self->wl_pool)
+    return FALSE;
+
   munmap (self->data, self->size);
+
   wl_shm_pool_destroy (self->wl_pool);
 
   self->wl_pool = NULL;
-- 
2.7.4

